File: src/074. Digit factorial chains/074.py
factorial = [1]
for d in range(1, 10):
    factorial.append(factorial[-1] * d)

def digit_fact(n):
    return sum([factorial[int(d)] for d in str(n)])

counter = 0
for n in range(10 ** 6):
    m = n
    sequence = set()
    while True:
        sequence.add(m)
        m = digit_fact(m)
        if m in sequence:
            break
    if len(sequence) == 60:
        counter += 1

print(counter)

File: src/074. Digit factorial chains/074.cpp
#include <iostream>
#include <unordered_set>
#include <vector>

std::vector<long long> factorial;
void init_factorial();
long long digit_fact(long long n);

int main()
{
    init_factorial();
    int counter = 0;
    for (long long n = 1; n <= 1000000LL; ++n)
    {
        long long m = n;
        std::unordered_set<long long> sequence;
        do
        {
            sequence.insert(m);
            m = digit_fact(m);
        } while (sequence.find(m) == sequence.end());
        if (sequence.size() == 60)
        {
            ++counter;
        }
    }
    std::cout << counter << std::endl;
}

void init_factorial()
{
    factorial.push_back(1);
    for (long long d = 1; d <= 9; ++d)
    {
        factorial.push_back(factorial[d - 1] * d);
    }
}

long long digit_fact(long long n)
{
    long long sum = 0;
    while (n > 0)
    {
        sum += factorial[n % 10];
        n /= 10;
    }
    return sum;
}
